test/common/network/dns_impl_test.cc

```diff
@@ -61,13 +61,14 @@ class TestDnsServerQuery {
       buffer_.add(data);
       while (true) {
         if (size_ == 0) {
-          uint16_t size_n;
-          if (buffer_.length() < sizeof(size_n)) {
+          if (buffer_.length() < sizeof(uint16_t)) {
             // If we don't have enough bytes to determine size, wait until we do.
             return;
           }
-          size_n = *static_cast<uint16_t*>(buffer_.linearize(sizeof(size_n)));
-          buffer_.drain(sizeof(size_n));
+          uint16_t size_n;
```

Comment:
Please undo the bit where you s/size_n/uint16_t in the `sizeof` statements - it's better form in general to use the actual variable rather than type name as you get the safety of having the compiler figure out the size.